lib/streamlit/elements/camera_input.py

```diff
@@ -56,17 +56,19 @@ def _get_file_recs_for_camera_input_widget(
     if len(uploaded_file_info) == 0:
         return []
 
-    active_file_urls = [f.file_url for f in uploaded_file_info]
+    active_file_ids = [f.file_id for f in uploaded_file_info]
 
     # Grab the files that correspond to our active file IDs.
     return ctx.uploaded_file_mgr.get_files(
         session_id=ctx.session_id,
-        file_urls=active_file_urls,
+        file_ids=active_file_ids,
     )
 
 
 @dataclass
 class CameraInputSerde:
+    file_delete_urls: Dict[str, str] = field(default_factory=dict)
```

Comment:
Hm, thinking about this one more. I think there's probably no point in trying to not expose this to the app developer as it has to be sent to the frontend anyway. Can reevaluate later, though, and maybe remove this if we don't think it's worth the additional complexity